lib/senkyoshi/models/file_resource.rb

```diff
@@ -19,5 +19,20 @@ def self.from(xml, pre_data, _resource_xids = nil)
     def iterate_xml(_xml, _pre_data)
       self
     end
+
+    def create_module(course, module_item)
+      course.canvas_modules ||= []
+      cc_module = course.canvas_modules.
```

Comment:
This detect is still repeated in the code. My thoughts are to make an instance variable on a course that holds the canvas_modules. Make an `attr_accessible` for it. In the initializer for a course set it to an empty array. And write a method
```ruby
def master_module
  @canvas_modules.detect { |a| a.title == MASTER_MODULE }
end
```

now you can say `cc_module = course.master_module` here and elsewhere.